### PR TITLE
Title: Confirmation of code

### PR DESCRIPTION
**Pull request by blacdev**
Body: The is is an attempt to write the endpoint.

 Changes to be committed:
	modified:   controller/base.go
	modified:   server/server.go

### CODE REVIEW

**Comment by blacdev** [comment]
@thealamu review the code

File: controller/base.go [hunk 1]
```diff
@@ -1,10 +1,85 @@
...
+
+	if username == "" || password == "" {
+		return c.String(http.StatusBadRequest, "email and password are required")
+	}
+
+	if !passwordMatch {
+		return c.String(http.StatusBadRequest, "password is invalid")
+	}
+
+	return c.String(http.StatusOK, "username: "+username)
```
Comment (tfalamu) [inline controller/base.go]: Let's have it return a JSON

**Review by tfalamu — COMMENTED** [review]
Please add unit tests for functions like the password hasher and username generator.

**Comment by tfalamu** [comment]
@blacdev could you add unit tests? For functions like the Password hasher and user name generator ...

**Comment by blacdev** [comment]
I'll attempt it

**Comment by blacdev** [comment]
@thealamu the tests have been written to cover the username generator and password hashing function

**Pull request merged**